packages/utils/src/ecdh.ts

```diff
@@ -0,0 +1,18 @@
+import { trimLeading0x } from '@celo/base/lib/address'
+import { createECDH } from 'crypto'
+
+export function computeSharedSecret(privateKey: string, publicKey: string): Buffer {
+  const ecdh = createECDH('secp256k1')
+  ecdh.setPrivateKey(Buffer.from(trimLeading0x(privateKey), 'hex'))
+  return ecdh.computeSecret(Buffer.from(ensurePublicKeyPrefix(publicKey), 'hex'))
+}
+
+function ensurePublicKeyPrefix(publicKey: string): string {
```

Comment:
All of this assumes the key is already in compressed form. We either need to make sure all callers of this function pass in a compressed key, or we compress it ourselves before adding the `04` prefix. It's unclear to me from the `crypto` module docs what encodings the `computeSecret` function will accept, but that may be worth verifying. (Additional context: The `Accounts` contracts does not specify whether users should use compressed or uncompressed forms for their key)

Reply:
To be clear, the 04 prefix is for uncompressed keys. What we'll do is have an `ensureCompressed` function so whatever is coming back from the blockchain / elsewhere ends us how we need it.

Reply:
Our ECIES `Encrypt` function takes non prefixed uncompressed public keys, and appends the prefix itself, I don't think we want to change that in this PR. So we now have

`ensureCompressed`
`ensureUncompressed`
`trimUncompressedPrefix`
`ensureUncompressedPrefix`